File: Programming-Basics-With-JavaScript/Homeworks-And-Labs/05.While-Loop/02-exercise/07.moving.js
function moving(input) {

    const widthFreeSpace = Number(input[0]);
    const lengthFreeSpace = Number(input[1]);
    const hightFreeSpace = Number(input[2]);
    const volumeFreeSpace = widthFreeSpace * lengthFreeSpace * hightFreeSpace;

    let index = 3;
    let current = input[index++];
    let filledSpace = 0;
    let isHaveFreeSpace = true;

    while (current !== "Done") {
        let currentBoxes = Number(current);
        // Бележка: Един кашон е с точни размери: 1m. x 1m. x 1m.
        filledSpace += currentBoxes;
        if (filledSpace >= volumeFreeSpace) {
            isHaveFreeSpace = false;
            break;
        }

        current = input[index++];
    }

    const difference = Math.abs(volumeFreeSpace - filledSpace);
    if (!isHaveFreeSpace) {
        console.log(`No more free space! You need ${difference} Cubic meters more.`);
    } else {
        console.log(`${difference} Cubic meters left.`);
    }
}
moving(["10",
"1",
"2",
"4",
"6",
"Done"]);

File: Python_Basics/Homeworks-And-Labs/05.while_loop/02.exercise/07.moving.py
width_free_space = int(input())
length_free_space = int(input())
hight_free_space = int(input())
volume_free_space = width_free_space * length_free_space * hight_free_space

current = input()
filled_space = 0
is_have_free_space = True

while current != "Done":
    current_boxes = int(current)
    #  Бележка: Един кашон е с точни размери: 1m. x 1m. x 1m.
    filled_space += current_boxes
    if filled_space >= volume_free_space:
        is_have_free_space = False
        break
        
    current = input()

difference = abs(volume_free_space - filled_space)
if not is_have_free_space:
    print(f'No more free space! You need {difference} Cubic meters more.')
else:
    print(f'{difference} Cubic meters left.')
